sandboxify.c

```diff
@@ -25,89 +27,144 @@ static void usage(const char *prog)
 
 int main(int argc, char **argv)
 {
+	int pipefds[2];
+	char data = '\0';
 	pid_t child = 0;
+	pid_t parent = getpid();
 
 	if (argc < 2) {
 		usage(basename(argv[0]));
 		return EXIT_FAILURE;
 	}
 
+	/* Could use pipe2 */
+	if (pipe(pipefds)) {
+		perror("Failed to setup pipes");
+		return EXIT_FAILURE;
+	}
+	fcntl(pipefds[0], F_SETFD, FD_CLOEXEC);
+	fcntl(pipefds[1], F_SETFD, FD_CLOEXEC);
+
 	child = fork();
 	if (-1 == child) {
 		perror("failed to spawn a child process");
 		return EXIT_FAILURE;
 	}
 
 	if (child) {
+		/* We created the child process */
 		int status = 0;
-		pid_t waitp = waitpid(child, &status, 0);
-
-		/* this should be executed after the first SIGSTOP from the child */
-		if (ptrace(PTRACE_SETOPTIONS, child, NULL, (TRACE_OPTS | PTRACE_O_SUSPEND_SECCOMP | PTRACE_O_EXITKILL))) {
-			perror("failed to suspend seccomp filters for the child");
-			/* PTRACE_O_EXITKILL was not yet set, so kill the child manually */
-			if (kill(child, SIGKILL)) {
-				perror("failed to kill the child process");
-			}
+		pid_t waitp;
+
+		if (close(pipefds[1])) {
+			perror("Closing pipe end");
+			return EXIT_FAILURE;
+		}
+
+		/* Allow the child to use PTRACE on us */
+		if (prctl(PR_SET_PTRACER, child, 0, 0, 0) < 0) {
+			perror("ptrctl");
```

Comment:
some meaningful error message here, like `failed to nominate the tracer for the parent`?